src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+                {
+                    CompletionHandlerConversion => await HandleConversionChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    CompletionHandlerIndexer => await HandleIndexerChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    _ => null,
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var symbols = await SymbolCompletionItem.GetSymbolsAsync(item, document, cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var symbol = symbols.Single() as IMethodSymbol;
+            var convertToType = symbol!.ReturnType;
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindTokenOnLeftOfPosition(position);
+            var expression = GetExpressionOfInvocation(token);
+            if (expression is { })
+            {
+                var typeName = convertToType.ToMinimalDisplayString(await document.ReuseExistingSpeculativeModelAsync(expression.SpanStart, cancellationToken).ConfigureAwait(false), expression.SpanStart);
+
+                // expr. -> ((type)expr).
+                var newNode =
+                    ParenthesizedExpression(
+                        CastExpression(IdentifierName(typeName), expression.WithoutTrivia()));
+                var newNodeText = newNode.ToFullString();
+                var expressionSpan = expression.Span;
+                var textChange = new TextChange(expressionSpan, newNodeText);
+                var newPosition = position + (newNodeText.Length - expressionSpan.Length);
+                return CompletionChange.Create(textChange, newPosition);
+            }
+
+            return null;
+        }
+
+        private static async Task<CompletionChange?> HandleIndexerChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var token = root.FindTokenOnLeftOfPosition(position);
+            if (token.IsKind(SyntaxKind.DotToken))
+            {
+                var newPosition = token.Span.End;
+                return CompletionChange.Create(new TextChange(token.Span, "[]"), newPosition);
+            }
+
+            return null;
+        }
+    }
+}
```

Comment:
so, i'm basically seeing extremely little sharing between conversions/indexers.  i would just have those be separate providers :)

Reply:
Yes. I was thinking about splitting it up too. But if we go on with operators, we may face some common problems. Prefix operators share some commons with explicit conversions. and binary and postfix share some logic with indexers. I would like to keep it together for the moment and come up with classes (maybe even a class hierarchy) when the operators are done and the completion handling (sorting, filtering, commit behavior, etc.) is properly working.

Reply:
I did split the provider into `ConversionCompletionProvider`, `IndexerCompletionProvider` and `OperatorCompletionProvider` now with a common base class `OperatorIndexerCompletionProviderBase` in a folder named "OperatorsAndIndexer".